app/input/multiple_dates_with_types_input.rb

```diff
@@ -1,28 +1,22 @@
-class MultipleDatesWithTypesInput < MultiValueInput
+class MultipleDatesWithTypesInput < AMS::MultiTypedInput
 
-  def build_field(value, index)
-    date_types_service = DateTypesService.new
-    date_type_choices = date_types_service.select_all_options
-
-    input_dom_id_prefix = "#{object_name}_#{attribute_name}_#{index}"
-
-    select_input_html_options = input_html_options.dup.merge({
-      name: "#{@builder.object_name}[date_type][]",
-      id: input_dom_id_prefix + "_type"
-    })
+  def text_input_html_options(value, index)
+    options = super
+    options[:pattern] = AMS::NonExactDateService.regex.to_s
+    options
+  end
 
-    date_input_html_options = input_html_options.dup.merge({
-      name: "#{@builder.object_name}[date_value][]",
-      value: value[1],
-      autocomplete:"off",
-      id: input_dom_id_prefix + "_input"
-    })
+  def type_choices
+    date_types_service = DateTypesService.new
+    choices = date_types_service.select_all_options
```

Comment:
No need for assignment here, can just return `date_types_service.select_all_options`